mainWindow.py

```diff
@@ -67,11 +68,19 @@ def openFile(self):
         
         fileOpener = fileOpenDialog.fileOpenDialog()
         fileOpener.openFileNameDialog()
+
         if fileOpener.fileAName != "":
             fileOpener.openFileNameDialog()
 
         fileA = fileOpener.fileAName
         fileB = fileOpener.fileBName
+
+        if not utilities.file_writable(fileA):
+            QMessageBox.about(self, "Error", os.path.basename(fileA) + " is not writable")
+            return
+        if not utilities.file_writable(fileB):
+            QMessageBox.about(self, "Error", os.path.basename(fileB) + " is not writable")
+            return
```

Comment:
These lines only work if the file is opened via file open operation; if passed via command line it does not work. Any file sanitation should be in the fileIO class. All file opens pass through this class before being opened. Also, just prompt the user, DO NOT abort merge. Let them know its write protected but allow it to be diffed and compared in the event they just wish to view differences but don't wish to actually merge or save changes.